dask/sizeof.py

```diff
@@ -135,44 +135,78 @@ def register_pandas():
     import numpy as np
     import pandas as pd
 
-    def object_size(x):
-        if not len(x):
+    from dask.dataframe._compat import dtype_eq
+
+    def object_size(*xs):
+        if not xs:
+            return 0
+        ncells = sum(len(x) for x in xs)
+        if not ncells:
             return 0
-        sample = np.random.choice(x, size=20, replace=True)
-        sample = list(map(sizeof, sample))
-        return sum(sample) / 20 * len(x)
+
+        # Deduplicate Series of references to the same objects,
+        # e.g. as produced by read_parquet
+        unique_samples = {}
+        for x in xs:
+            sample = np.random.choice(x, size=100, replace=True)
+            for i in sample.tolist():
+                unique_samples[id(i)] = i
```

Comment:
I'm guessing that this is fast, but can I ask you to verify briefly that this doesn't significantly increase the time cost of `sizeof` for series or dataframe objects?  This has been a surprising bottleneck in thepast.

Reply:
On one partition of the NYC taxi database:

before: 195 µs ± 2.48 µs
after: 343 µs ± 4.09 µs

Note that I increased the number of samples from 20 to 100 for better accuracy.

Reply:
I'm not thrilled to see any increase, but this is probably low enough not to arise frequently on profiles.  For context, this often comes up in task-based shuffle, where a lot of what we do is create lots of little small dataframes with getitem calls.  

Anyway, thank you for doing the microbenchmark.  Happy to relax my previous concern.